river/utils/estimator_checks.py

```diff
@@ -130,6 +136,30 @@ def check_shuffle_features_no_impact(model, dataset):
         shuffled.learn_one(x_shuffled, y)
 
 
+def check_emerging_features(model, dataset):
+    """The model should work fine when new features appear."""
+
+    x, _ = next(iter(dataset))
+    features = list(x.keys())
+
+    for x, y in dataset:
+        random.shuffle(features)
+        model.learn_one({i: x[i] for i in features[:-3]}, y)  # drop 3 features at random
+        model.predict_one(x)
+
+
+def check_disappearing_features(model, dataset):
+    """The model should work fine when features disappear."""
+
+    x, _ = next(iter(dataset))
+    features = list(x.keys())
+
+    for x, y in dataset:
+        random.shuffle(features)
+        model.learn_one(x, y)
+        model.predict_one({i: x[i] for i in features[:-3]})  # drop 3 features at random
```

Comment:
Wondeful!

Reply:
I think so too! We should get in the habit of writing these kinds of tests, it will make our code much more fool proof.

Reply:
Totally agreed!